### PR TITLE
Drop Python 2.7/3.5 in tests

### PR DESCRIPTION
**Pull request by jpivarski**
I noticed that a Python 3.5/pip version conflict is raising a warning in GitHub Actions. It's probably time to make the minimum version 3.6. I also updated the middle version.

You know what? Drop Python 2.7, too.

This is a proposal: I won't merge this until I get feedback.

### CODE REVIEW

**Review by eduardo-rodrigues — APPROVED** [review]
I did not go and check all org packages but believe that those using scikit-hep-testdata have moved past Python 3.6 by now. I believe this is a fair update.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,9 +14,8 @@ jobs:
       fail-fast: false
       matrix:
         python-version:
-        - '2.7'
-        - '3.5'
-        - '3.7'
+        - '3.6'
+        - '3.8'
```
Comment (matthewfeickert) [inline .github/workflows/ci.yml:18]: @jpivarski is there a need to drop Python 3.7 here as well? If not, might as well keep it.

```suggestion
        - '3.7'
        - '3.8'
```

**Review by matthewfeickert — APPROVED** [review]
LGTM. 👍

Shall we merge?

**Comment by jpivarski** [comment]
Okay, so "3.6, 3.7, 3.8, ???, 3.10" is an odd list of Python versions to check.

For this package, which just delivers sample files, and is implemented in pure Python, I think we only really need to test the minimum and maximum Python versions: "3.6, 3.10".

**Comment by jpivarski** [comment]
I think I'm going to make that change and then merge. Any objections, Matt? You're the second pair of eyes.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -14,9 +14,9 @@ jobs:
       fail-fast: false
       matrix:
         python-version:
-        - '2.7'
-        - '3.5'
+        - '3.6'
         - '3.7'
+        - '3.8'
```
Comment (jpivarski) [inline .github/workflows/ci.yml:19]: This is that suggestion I made in the comment above.

```suggestion
        - '3.6'
```

Comment (jpivarski) [reply .github/workflows/ci.yml:19]: 3.10 is on the following line, after this diff.

Comment (jpivarski) [reply .github/workflows/ci.yml:19]: The alternative is to do every Python version, "3.6, 3.7, 3.8, 3.9, 3.10". Either we keep all the Python versions in the supported range or we do min-max, but just excluding 3.9 is weird.

Comment (matthewfeickert) [reply .github/workflows/ci.yml:19]: Oh I missed that Python 3.9 was skipped. IMO the CI takes less than 1 minute per Python at the moment, so I personally lean towards the "test explicitly every Python you claim to support". But at the same time I don't have any real problem to min-max-ing it.

So I'll defer to others, but if it was me I'd do 3.6, 3.7, 3.8, 3.9, 3.10.

Comment (jpivarski) [reply .github/workflows/ci.yml:19]: Okay, I'll do all the Python versions. Since they're fast, anyway!

Comment (jpivarski) [inline .github/workflows/ci.yml:19]:
```suggestion
        - '3.6'
        - '3.7'
        - '3.8'
        - '3.9'
```

**Comment by jpivarski** [comment]
Now they're all required and they all ran. I'm going to merge this.
